torch/_subclasses/fake_tensor.py

```diff
@@ -583,13 +583,23 @@ def infer_size(a, b):
         dimB = dimsB - 1 - offset
         sizeA = a[dimA] if dimA >= 0 else 1
         sizeB = b[dimB] if dimB >= 0 else 1
-        if not (sizeA == sizeB or sizeA == 1 or sizeB == 1):
+        from torch.fx.experimental.symbolic_shapes import definitely_true
+        if definitely_true(sizeA == 1):
```

Comment:
can you leave a comment here about how `definitely_true` case is different from `sizeB == 1` below? We don't expect unbacked symints to hit `sizeB == 1`?